terraform/ec2/mac/main.tf

```diff
@@ -0,0 +1,92 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT
+
+module "common" {
+  source = "../../common"
+}
+
+#####################################################################
+# Generate EC2 Key Pair for log in access to EC2
+#####################################################################
+
+resource "tls_private_key" "ssh_key" {
+  count     = var.ssh_key_name == "" ? 1 : 0
+  algorithm = "RSA"
+  rsa_bits  = 4096
+}
+
+resource "aws_key_pair" "aws_ssh_key" {
+  count      = var.ssh_key_name == "" ? 1 : 0
+  key_name   = "ec2-key-pair-${module.common.testing_id}"
+  public_key = tls_private_key.ssh_key[0].public_key_openssh
+}
+
+locals {
+  ssh_key_name        = var.ssh_key_name != "" ? var.ssh_key_name : aws_key_pair.aws_ssh_key[0].key_name
+  private_key_content = var.ssh_key_name != "" ? var.ssh_key_value : tls_private_key.ssh_key[0].private_key_pem
+}
+
+#####################################################################
+# Generate EC2 Instance and execute test commands
+#####################################################################
+
+data "template_file" "macos-userdata" {
+  template = <<EOF
+  #! /bin/bash
+  sudo softwareupdate --install-rosetta --agree-to-license
+  curl https://awscli.amazonaws.com/AWSCLIV2.pkg -o AWSCLIV2.pkg,
+  sudo installer -pkg AWSCLIV2.pkg -target /
+  EOF
+}
+
+resource "aws_instance" "cwagent" {
+  ami                         = data.aws_ami.latest.id
+  instance_type               = var.ec2_instance_type
+  key_name                    = local.ssh_key_name
+  iam_instance_profile        = data.aws_iam_instance_profile.cwagent_instance_profile.name
+  vpc_security_group_ids      = [data.aws_security_group.ec2_security_group.id]
+  user_data                   = data.template_file.macos-userdata.rendered
+  associate_public_ip_address = true
+  tenancy                     = "host"
+
+
+  metadata_options {
+    http_endpoint = "enabled"
+    http_tokens   = "required"
+  }
+
+  tags = {
+    Name = "cwagent-integ-test-ec2-${var.test_name}-${module.common.testing_id}"
+  }
+}
+
+resource "null_resource" "integration_test" {
+  connection {
+    type        = "ssh"
+    user        = var.user
+    private_key = local.private_key_content
+    host        = aws_instance.cwagent.public_ip
+    timeout     = "10m"
+  }
+
+  # Install agent binaries
+  provisioner "remote-exec" {
```

Comment:
We SSH into the EC2 Mac Instances and install the required binary packages for CWA, we will add the required validation step in the next PR.